### PR TITLE
Add OCaml 4.11.1 and update patch versions of older compiler versions

### PR DESCRIPTION
**Pull request by smorimoto**
Related PR: https://github.com/compiler-explorer/infra/pull/422

### CODE REVIEW

File: etc/config/ocaml.amazon.properties [hunk 1]
```diff
@@ -19,14 +19,18 @@ compiler.ocaml4081.semver=4.08.1
...
-compiler.ocaml4090.exe=/opt/compiler-explorer/ocaml-4.09.0/bin/ocamlopt
-compiler.ocaml4090.semver=4.09.0
-compiler.ocaml4090flambda.exe=/opt/compiler-explorer/ocaml-4.09.0-flambda/bin/ocamlopt
-compiler.ocaml4090flambda.semver=4.09.0-flambda
-compiler.ocaml4100.exe=/opt/compiler-explorer/ocaml-4.10.0/bin/ocamlopt
-compiler.ocaml4100.semver=4.10.0
-compiler.ocaml4100flambda.exe=/opt/compiler-explorer/ocaml-4.10.0-flambda/bin/ocamlopt
-compiler.ocaml4100flambda.semver=4.10.0-flambda
+compiler.ocaml4091.exe=/opt/compiler-explorer/ocaml-4.09.1/bin/ocamlopt
+compiler.ocaml4091.semver=4.09.1
```
Comment (partouf) [inline etc/config/ocaml.amazon.properties:23]: Please add `compiler.ocaml4091.alias=ocaml4090`

File: etc/config/ocaml.amazon.properties [hunk 2]
```diff
@@ -19,14 +19,18 @@ compiler.ocaml4081.semver=4.08.1
...
+compiler.ocaml4091flambda.exe=/opt/compiler-explorer/ocaml-4.09.1-flambda/bin/ocamlopt
+compiler.ocaml4091flambda.semver=4.09.1-flambda
```
Comment (partouf) [inline etc/config/ocaml.amazon.properties:25]: Please add `compiler.ocaml4091flambda.alias=ocaml4090flambda`

File: etc/config/ocaml.amazon.properties [hunk 3]
```diff
@@ -19,14 +19,18 @@ compiler.ocaml4081.semver=4.08.1
...
+compiler.ocaml4101.exe=/opt/compiler-explorer/ocaml-4.10.1/bin/ocamlopt
+compiler.ocaml4101.semver=4.10.1
```
Comment (partouf) [inline etc/config/ocaml.amazon.properties:27]: `compiler.ocaml4101.alias=ocaml4100`

File: etc/config/ocaml.amazon.properties [hunk 4]
```diff
@@ -19,14 +19,18 @@ compiler.ocaml4081.semver=4.08.1
...
+compiler.ocaml4101flambda.exe=/opt/compiler-explorer/ocaml-4.10.1-flambda/bin/ocamlopt
+compiler.ocaml4101flambda.semver=4.10.1-flambda
```
Comment (partouf) [inline etc/config/ocaml.amazon.properties:29]: compiler.ocaml4101flamda.alias=ocaml4100flambda

**Review by partouf — CHANGES_REQUESTED** [review]
Noted some things for aliases to the old compilers to keep old shared links working

**Comment by smorimoto** [comment]
@partouf Oops... Indeed! I fixed that now. And I've made a similar fix to the infrastructure side!

**Comment by partouf** [comment]
> @partouf Oops... Indeed! I fixed that now. And I've made a similar fix to the infrastructure side!

Thanks!

**Comment by smorimoto** [comment]
Thank you, too! Your review was really quick and accurate!

**Comment by partouf** [comment]
These are now live. Thanks again!

**Comment by smorimoto** [comment]
Yay! Thanks!

**Pull request merged**